src/state/user/actions.js

```diff
@@ -7,10 +7,107 @@ const setUser = payload => ({
   type: 'SET_USER',
 });
 
+const setMOCs = payload => ({
+  payload,
+  type: 'SET_MOCS',
+});
+
+const addLink = (payload) => ({
+  type: 'ADD_LINK',
+  payload,
+});
+
+const editLink = (payload) => ({
+  type: 'EDIT_LINK',
+  payload
+});
+
+const deleteLink = (payload) => ({
+  type: 'DELETE_LINK',
+  payload
+});
+
+export const setSelectedLink = (payload) => ({
+  type: 'SET_SELECTED_LINK',
+  payload
+})
+
 export const writeUserData = payload => (dispatch) => {
   firebasedb.ref(`users/${payload.uid}`).update({
     email: payload.email,
     username: payload.displayName,
   })
     .then(dispatch(setUser(payload)));
 };
+
+export const getUserMOCs = payload => (dispatch) => {
+  firebasedb.ref(`users/${payload.uid}/mocs`).once('value').then((snapshot) => {
+    const mocIds = Object.keys(snapshot.val());
+    Promise.all(mocIds.map((id) => {
+      return firebasedb.ref(`mocData/${id}`).once('value').then((snapshot) => {
+        let moc = snapshot.val();
+        let mocData = {
+          govtrack_id: moc.govtrack_id,
+          member_name: moc.displayName,
+        }
+        return mocData;
+      })
+    }))
+      .then((userMocData) => {
+        dispatch(setMOCs(userMocData))
+      })
+      .catch((err) => console.log(err))
+  })
+}
+
+/******************* MOC LINK API******************************* */
+export const setSelectedMember = payload  => ({
+  payload,
+  type: 'SET_SELECTED_MEMBER',
+});
+
+export const getSelectedMemberInfo = payload => (dispatch) => {
```

Comment:
this should be in the member-candidate branch of state

Reply:
both the member selection and links actions, right?

Reply:
No, the selected member should be in 'selections'. But if you're manipulating the moc data, then that should be in member-candidate.

Reply:
Ok, I've separated a number of actions from users/ branch to member-candidate/ branch. The selected member and links is in member-candidate for now - the MOC link actions come from those selected states (add, edit, delete - selectedMoc and selectedLink)